src/handlers/tests/ResponseHandlers.spec.ts

```diff
@@ -19,24 +19,103 @@ describe('ResponseHandlers', () => {
-        const httpError = {code: 'WRONG_UTENSIL', message: 'Use a spoon to eat the soup.'};
-        const errorResponse = new Response(JSON.stringify(httpError), {
-            status: 400,
-            headers: {
-                'X-Request-ID': 'DidyoueverhearthetragedyofDarthPlagueisTheWise?',
-            },
+    describe('when the status is not between 200 and 299', () => {
+        const buildResponseAndError = (httpError: object = {}, headers: HeadersInit = {}) => {
+            const errorResponse = new Response(JSON.stringify(httpError), {
+                status: 400,
+                headers,
+            });
+            return {errorResponse, httpError};
+        };
+
+        it('should return a promise rejected with the response body', async () => {
+            const {errorResponse, httpError} = buildResponseAndError();
+
+            let rejectedError: CoveoPlatformClientError | undefined;
+            try {
+                await handleResponse(errorResponse);
+            } catch (error) {
+                rejectedError = error;
+            }
+
+            expect(rejectedError).toMatchObject(httpError);
+            expect(rejectedError).toBeInstanceOf(CoveoPlatformClientError);
+        });
+
+        it('should include the HTTP status code in the error', async () => {
+            const {errorResponse} = buildResponseAndError();
+
+            let rejectedError: CoveoPlatformClientError | undefined;
+            try {
+                await handleResponse(errorResponse);
+            } catch (error) {
+                rejectedError = error;
+            }
+
+            expect(rejectedError?.status).toBe(400);
+        });
+
+        it('should include the `xRequestId` in the error', async () => {
+            const {errorResponse} = buildResponseAndError(
+                {},
+                {
+                    'X-Request-ID': 'DidyoueverhearthetragedyofDarthPlagueisTheWise?',
+                }
+            );
+            let rejectedError: CoveoPlatformClientError | undefined;
+            try {
+                await handleResponse(errorResponse);
+            } catch (error) {
+                rejectedError = error;
+            }
+
+            expect(rejectedError?.xRequestId).toBe('DidyoueverhearthetragedyofDarthPlagueisTheWise?');
+        });
+
+        it.each(['xRequestId', 'title', 'detail'])(
+            'should ultimately fallback `%s` to unknown as a string',
+            async (key) => {
+                const {errorResponse} = buildResponseAndError();
+                let rejectedError: CoveoPlatformClientError | undefined;
+                try {
+                    await handleResponse(errorResponse);
+                } catch (error) {
+                    rejectedError = error;
+                }
+
+                expect(rejectedError?.[key]).toBe('unknown');
+            }
+        );
```

Comment:
Personally I prefer to have a single test with 3 different expect than a `it.each`, I find it easier to read and debug. I'm not opposed to keep it that way if the majority prefer it that way, I just wanted to voice my concern

Ref. https://kentcdodds.com/blog/write-fewer-longer-tests

Reply:
Wouldn't it be the same to just add a new expect? E.g, `expect(rejectedError?.newKey).toBe('unknown');`

Reply:
IMHO, no:
1. It's more verbose
2. Most importantly, with Jest you cannot write custom 'fail text', so using the `it`/`test` to have a human-friendly text is important. (e.g. if you had one key or change the behaviour of a single key, it's easy to see directly from the test report what's wrong)

Reply:
I understand your points but IMHO 

1. Tests being more verbose is not a bad thing, I would even say that I prefer them that way. I think it reduces the cognitive load because everything is clear and easy to understand
2. I could rewrite your test like this and I feel like it would check all the requirements (easy to debug, human-friendly text, easy to read test report)

```typescript
it('fallbacks to "unknown" for standard error fields', async () => {
    const {errorResponse} = buildResponseAndError();
    let rejectedError: CoveoPlatformClientError | undefined;
    try {
        await handleResponse(errorResponse);
    } catch (error) {
        rejectedError = error;
    }

    expect(rejectedError?.xRequestId).toBe('unknown');
    expect(rejectedError?.title).toBe('unknown');
    expect(rejectedError?.detail).toBe('unknown');
});
```

Example of a report when I make a mistake

```bash
 FAIL   ts-check and test  src/handlers/tests/ResponseHandlers.spec.ts (5.734 s)
  ● ResponseHandlers › when the status is not between 200 and 299 › fallbacks to "unknown" for standard error fields

    expect(received).toBe(expected) // Object.is equality

    Expected: "unknown"
    Received: "oups"

      83 |
      84 |             expect(rejectedError?.xRequestId).toBe('unknown');
    > 85 |             expect(rejectedError?.title).toBe('unknown');
         |                                          ^
      86 |             expect(rejectedError?.detail).toBe('unknown');
      87 |         });
      88 |

      at Object.<anonymous> (handlers/tests/ResponseHandlers.spec.ts:85:42)
```

I think it's even easier to see where the error is than in your implementation. Instead of relying on the test title and the generic `expect(rejectedError?.[key]).toBe('unknown');` I just know that error is on the title field `expect(rejectedError?.title).toBe('unknown');`

Your PR might not be the best place to discuss this, and as I said previously it's mostly a preference and maybe a question of habit. Since I mostly do UI/frontend tests I'm probably biased

I approved the PR and I'm not against merging it in the current state, it's really a minor comment